From bf916aafc3df7bc1f6bc4da55a59f4876a1ed647 Mon Sep 17 00:00:00 2001
From: Brad Geltz <brad.geltz@intel.com>
Date: Wed, 5 Aug 2020 13:34:15 -0700
Subject: [PATCH 3/8] Link w/GEOPM
#
#  Copyright (c) 2015 - 2021, Intel Corporation
#
#  Redistribution and use in source and binary forms, with or without
#  modification, are permitted provided that the following conditions
#  are met:
#
#      * Redistributions of source code must retain the above copyright
#        notice, this list of conditions and the following disclaimer.
#
#      * Redistributions in binary form must reproduce the above copyright
#        notice, this list of conditions and the following disclaimer in
#        the documentation and/or other materials provided with the
#        distribution.
#
#      * Neither the name of Intel Corporation nor the names of its
#        contributors may be used to endorse or promote products derived
#        from this software without specific prior written permission.
#
#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
#  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
#  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
#  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
#  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
#  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
#  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
#  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
#  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
#  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY LOG OF THE USE
#  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
#

- Also use default compilers (not Intel wrappers)

Signed-off-by: Brad Geltz <brad.geltz@intel.com>
---
 test/example1/makenek-intel | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/test/example1/makenek-intel b/test/example1/makenek-intel
index a5cce80..82ec388 100755
--- a/test/example1/makenek-intel
+++ b/test/example1/makenek-intel
@@ -6,10 +6,10 @@
 SOURCE_ROOT="../../src"
 
 # Fortran compiler
-F77="mpiifort"
+F77="${MPIF77} ${GEOPM_FFLAGS}"
 
 # C compiler
-CC="mpiicc"
+CC="${MPICC}"
 
 # pre-processor symbol list
 # (set PPLIST=? to get a list of available symbols)
@@ -29,10 +29,10 @@ PPLIST="TIMERS CGTIMERS"
 #USR="foo.o"
 
 # linking flags
-USR_LFLAGS="-qopenmp -mcmodel=medium -shared-intel"
+USR_LFLAGS="-qopenmp -mcmodel=medium -shared-intel ${GEOPM_LDFLAGS} ${GEOPM_LDLIBS} ${GEOPM_FORTRAN_LDLIBS}"
 
 # generic compiler flags
-#G="-g"
+G="-DGEOPM -I${GEOPM_ROOT}/include"
 
 # optimization flags
 OPT_FLAGS_STD="-qopenmp -O3 -g -xCORE-AVX2 -mcmodel=medium -shared-intel"
-- 
2.23.0

